tgui/packages/tgui/interfaces/MarionetteRemote.js

```diff
@@ -0,0 +1,163 @@
+                    />
+                    <Button
+                      onClick={() => act("set_command", { new_command: "shock" })}
+                      content="Shock"
+                      selected={selected_command === "shock"}
+                    />
+                  </LabeledList.Item>
+                  <LabeledList.Item key="action_heat" label="Heat Per Action">
+                    {selected_command === "shock" || selected_command === "drop" ? "HIGH"
+                      : selected_command === "step" ? "LOW" : "MEDIUM"}
+                  </LabeledList.Item>
+                </LabeledList>
+              </Flex.Item>
+            </Flex>
+          </Section>
+          <Section title="Implants" buttons={(
+            <Button
+              icon="rotate"
+              content="Refresh"
+              onClick={() => act('ping_all')}
+            />
+          )}>
+            {mapImplants(act, entered_data, selected_command, implants)}
+          </Section>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const tooltipForStatus = (status) => {
+  switch (status) {
+    case "IDLE":
+      return "This implant is not located inside a living being.";
+    case "ACTIVE":
+      return "This implant is inside a living being and ready to accept signals.";
+    case "WAITING...":
+      return "Awaiting ping response...";
+    case "DANGER":
+      return "This implant is dangerously hot. Further short-term use will likely cause it to break.";
+    case "NO RESPONSE":
+      return "This implant is not responding to pings. It could have been destroyed, or it could just be far away.";
+    case "BURNED OUT":
+      return "This implant has been rendered permanently unusable by overuse and can be removed from the tracking list.";
+    default:
+      return "Unknown.";
+  }
+};
+
+const mapImplants = (act, entered_data, selected_command, implants) => {
+  if (!implants || !implants.length)
+  { return (<i>No implants detected.</i>); }
+  return (
+    <LabeledList>
+      {implants.map(implant => (
+        <LabeledList.Item
+          key={implant.address}
+          label={implant.address}
+          buttons={(
+            <>
+              <Button
+                icon="info"
+                tooltip={tooltipForStatus(implant.status)}
+              />
+              <Button
+                icon="satellite-dish"
+                content="Ping"
+                onClick={() => act('ping', { address: implant.address })}
+                disabled={implant.status === "BURNED OUT"}
+              />
+              <Button
+                icon="envelope"
+                content="Activate"
+                onClick={() => act('activate', { address: implant.address, packet_data: entered_data, packet_command: selected_command })}
+                disabled={implant.status === "BURNED OUT"}
+              />
+              <Button.Confirm
+                icon="x"
```

Comment:
A tooltip would be useful here. What does this do, destory the implant? unlink it?